===================================================================
RCS file: /cvsroot/lib3ds/lib3ds/lib3ds/io.c,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- lib3ds/io.c	2001/07/11 13:47:35	1.3
+++ lib3ds/io.c	2005/12/30 10:24:44	1.4
@@ -17,7 +17,7 @@
  * along with  this program;  if not, write to the  Free Software Foundation,
  * Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  *
- * $Id: io.c,v 1.3 2001/07/11 13:47:35 jeh Exp $
+ * $Id: io.c,v 1.4 2005/12/30 10:24:44 madmac Exp $
  */
 #define LIB3DS_EXPORT
 #include <lib3ds/io.h>
@@ -247,16 +247,17 @@
 Lib3dsFloat
 lib3ds_io_read_float(Lib3dsIo *io)
 {
+  union f_and_dw { Lib3dsDword dwval; Lib3dsFloat fval;};
   Lib3dsByte b[4];
-  Lib3dsDword d;
+  union f_and_dw d;
 
   ASSERT(io);
   lib3ds_io_read(io, b, 4);
-  d=((Lib3dsDword)b[3] << 24) |
+  d.dwval=((Lib3dsDword)b[3] << 24) |
     ((Lib3dsDword)b[2] << 16) |
     ((Lib3dsDword)b[1] << 8) |
     ((Lib3dsDword)b[0]);
-  return(*((Lib3dsFloat*)&d));
+  return d.fval;
 }
 
 
